bundles/AdminBundle/Session/Handler/AdminSessionHandler.php

```diff
@@ -273,4 +272,23 @@ public function writeClose()
             ]);
         }
     }
+
+    /**
+     * @return bool
+     */
+    private function shouldWriteAndClose(): bool
+    {
+        if ($this->canWriteAndClose === null) {
+            $this->canWriteAndClose = false;
+            $request = $this->requestHelper->getMainRequest();
+            if (
+                $this->matchesPimcoreContext($request, PimcoreContextResolver::CONTEXT_ADMIN) ||
+                $this->requestHelper->isFrontendRequestByAdmin($request)
+            ) {
+                $this->canWriteAndClose = true;
+            }
+        }
+
+        return $this->canWriteAndClose;
+    }
```

Comment:
What about writing it like this? 
For me this is more readable:

```suggestion
    private function shouldWriteAndClose(): bool
    {
        return $this->canWriteAndClose ??= $this->isAdminRequest($this->requestHelper->getMainRequest());
    }

    /**
     * @return bool
     */
    private function isAdminRequest(Request $request): bool
    {
        return $this->matchesPimcoreContext($request, PimcoreContextResolver::CONTEXT_ADMIN)
            || $this->requestHelper->isFrontendRequestByAdmin($request);
    }
```

Reply:
and reusable as well 😉 thanks!